plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -93,6 +97,8 @@ Transaction newCreateTableTransaction(
 
     String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName);
 
+    String latestMetadataLocation(ConnectorSession session, String tableLocation);
```

Comment:
I am not very sure about this change. If it makes sense to add this method to the interface. Please suggest.
CC: @alexjo2144 | @findinpath | @findepi

Reply:
Can't the caller just use `IcebergUtil` directly?

Reply:
To get the metadata location, we need `TrinoFileSystem` object that is available in catalog. So if we want to call `IcebergUtil` directly then we have to call it from catalog class (`TrinoHiveCatalog/TrinoGlueCatalog`). And in this case, we won't able to get the metadata location before calling `registerTable` see https://github.com/trinodb/trino/pull/14375#discussion_r987253599

Reply:
The RegisterTableProcedure can take fileSystemFactory.
Then we don't need this call to go through catalog, especially that the result doesn't need to -- and must not be -- catalog-dependent. It's just supposed to find a file on the fileystem, and catalog has nothing to do with this.

Reply:
Removed this method. As mentioned by @findepi, We are not getting fileSystemFactory in `RegisterTableProcedure`.